test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs

```diff
@@ -437,6 +437,35 @@ public void DthCompilation_ChangeConfiguration(DnxSdk sdk)
             }
         }
 
+        [Theory, TraceTest]
+        [MemberData(nameof(DnxSdks))]
+        public void DthStartup_EnumerateProjectContexts(DnxSdk sdk)
+        {
+            // arrange
+            using (var server = sdk.Dth.CreateServer())
+            using (var client = server.CreateClient())
+            {
+                var solution = TestProjectsRepository.EnsureRestoredSolution("DthTestProjects");
+                var projects = new Project[]
+                {
+                    solution.GetProject("EmptyLibrary"),
+                    solution.GetProject("UnresolvedPackageSample"),
+                    solution.GetProject("EmptyConsoleApp")
+                };
+
+                var contexts = projects.ToDictionary(proj => proj.ProjectDirectory,
+                                                     proj => client.Initialize(proj.ProjectDirectory));
+                client.DrainMessage(21);
```

Comment:
Limits the number of messages to be collected/drained. This speeds up the tests.